src/impl_aarch64_windows.c

```diff
@@ -0,0 +1,150 @@
+  LINE(AARCH64_FRINT, frint, , , )           \
+  LINE(AARCH64_SVEI8MM, svei8mm, , , )       \
+  LINE(AARCH64_SVEF32MM, svef32mm, , , )     \
+  LINE(AARCH64_SVEF64MM, svef64mm, , , )     \
+  LINE(AARCH64_SVEBF16, svebf16, , , )       \
+  LINE(AARCH64_I8MM, i8mm, , , )             \
+  LINE(AARCH64_BF16, bf16, , , )             \
+  LINE(AARCH64_DGH, dgh, , , )               \
+  LINE(AARCH64_RNG, rng, , , )               \
+  LINE(AARCH64_BTI, bti, , , )               \
+  LINE(AARCH64_MTE, mte, , , )               \
+  LINE(AARCH64_ECV, ecv, , , )               \
+  LINE(AARCH64_AFP, afp, , , )               \
+  LINE(AARCH64_RPRES, rpres, , , )
+#define INTROSPECTION_PREFIX Aarch64
+#define INTROSPECTION_ENUM_PREFIX AARCH64
+#include "define_introspection.inl"
+
+////////////////////////////////////////////////////////////////////////////////
+// Implementation.
+////////////////////////////////////////////////////////////////////////////////
+
+#include <stdbool.h>
+
+#include "internal/windows_utils.h"
+
+#ifdef CPU_FEATURES_MOCK_CPUID_AARCH64
+extern bool GetWindowsIsProcessorFeaturePresent(DWORD);
+extern WORD GetWindowsNativeSystemInfoProcessorRevision();
+#else  // CPU_FEATURES_MOCK_CPUID_AARCH64
+static bool GetWindowsIsProcessorFeaturePresent(DWORD dwProcessorFeature) {
+  return IsProcessorFeaturePresent(dwProcessorFeature);
+}
+
+static WORD GetWindowsNativeSystemInfoProcessorRevision() {
+  SYSTEM_INFO system_info;
+  GetNativeSystemInfo(&system_info);
+  return system_info.wProcessorRevision;
+}
+#endif
+
+static const Aarch64Info kEmptyAarch64Info;
+
+Aarch64Info GetAarch64Info(void) {
+  Aarch64Info info = kEmptyAarch64Info;
+  info.revision = GetWindowsNativeSystemInfoProcessorRevision();
+  info.features.fp =
+      GetWindowsIsProcessorFeaturePresent(PF_ARM_VFP_32_REGISTERS_AVAILABLE);
+  info.features.asimd =
+      GetWindowsIsProcessorFeaturePresent(PF_ARM_NEON_INSTRUCTIONS_AVAILABLE);
+  info.features.crc32 = GetWindowsIsProcessorFeaturePresent(
+      PF_ARM_V8_CRC32_INSTRUCTIONS_AVAILABLE);
+  info.features.asimddp =
+      GetWindowsIsProcessorFeaturePresent(PF_ARM_V82_DP_INSTRUCTIONS_AVAILABLE);
+  info.features.jscvt = GetWindowsIsProcessorFeaturePresent(
+      PF_ARM_V83_JSCVT_INSTRUCTIONS_AVAILABLE);
+  info.features.lrcpc = GetWindowsIsProcessorFeaturePresent(
+      PF_ARM_V83_LRCPC_INSTRUCTIONS_AVAILABLE);
+
+  // According to documentation Arm Architecture Reference Manual for A-profile
+  // architecture. A2.3 The Armv8 Cryptographic Extension. The Armv8.0
+  // Cryptographic Extension provides instructions for the acceleration of
+  // encryption and decryption, and includes the following features: FEAT_AES,
+  // FEAT_PMULL, FEAT_SHA1, FEAT_SHA256.
+  // see: https://developer.arm.com/documentation/ddi0487/latest
+  bool is_crypto_available = GetWindowsIsProcessorFeaturePresent(
+      PF_ARM_V8_CRYPTO_INSTRUCTIONS_AVAILABLE);
+  info.features.aes = is_crypto_available;
```

Comment:
Unfortunately, these are all features for armv8+ (Windows 11 SDK 10.0.22621.0):
```C
#define PF_ARM_V8_INSTRUCTIONS_AVAILABLE            29   
#define PF_ARM_V8_CRYPTO_INSTRUCTIONS_AVAILABLE     30   
#define PF_ARM_V8_CRC32_INSTRUCTIONS_AVAILABLE      31   
#define PF_ARM_V81_ATOMIC_INSTRUCTIONS_AVAILABLE    34   
#define PF_ARM_V82_DP_INSTRUCTIONS_AVAILABLE        43   
#define PF_ARM_V83_JSCVT_INSTRUCTIONS_AVAILABLE     44   
#define PF_ARM_V83_LRCPC_INSTRUCTIONS_AVAILABLE     45   
```

In Microsoft documentation `PF_ARM_V8_CRYPTO_INSTRUCTIONS_AVAILABLE` described that: Arm processor implements the Arm v8 extra cryptographic instructions (i.e. AES, SHA1 and SHA2).
My understanding is that windows reports all armv8.0 extra cryptographic instructions or nothing.

ref: https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-isprocessorfeaturepresent

SIMD/FP must be implemented on all armv8 implementations, but implementations targeting specialized markets may support the following combinations:
* No NEON or floating-point.
* Full floating-point and SIMD support with exception trapping.
* Full floating-point and SIMD support without exception trapping.

ref: https://developer.arm.com/documentation/den0024/a/AArch64-Floating-point-and-NEON

So, I used, just in case, `PF_ARM_VFP_32_REGISTERS_AVAILABLE`, `PF_ARM_NEON_INSTRUCTIONS_AVAILABLE` for `asimd` and `fp`

Reply:
I will document it. About support of armv8.0 crypto instructions I see that in llvm host.cpp: https://llvm.org/doxygen/Host_8cpp_source.html#l01829 also reports all or nothing, so I think it will be enough to document this that we report in case if all crypto features armv8.0 are supported